### ISSUE TITLE
x86_64 build using crypto acceleration(ASM_X86) produces SHA extension instructions not available on many microarchitectures?

### ISSUE THREAD

**Issue by atraczyk** [ShiftMediaProject/gnutls#3]
The 64bit versions of the .asm files produced from the .s files have SHA extention instructions in them, which compile but don't run on the Broadwell-U system I work on. From what I've read on wikipedia and in Intel's docs, it seems these extensions are only available on Goldmont core systems which were unveiled in April. Disabling accelerated crypto is the work around I can find for now.

**Comment by Sibras** [comment]
That shouldnt happen as the code that uses sha instructions wont do so without checking for the returned flags from cpuid first. I looked at the asm and its definitely only running sha instructions when the cpu returns support for it.
When the culprit functions are run what is the status of the "gnutls_x86_cpuid_s" array? Knowing what your cpu is returning may help in identifying the problem.

**Comment by atraczyk** [comment]
These are the values in the "gnutls_x86_cpuid_s" array just after "gnutls_cpuid" returns:
[0x00000000]    0x03100800
[0x00000001]    0xfed87383
[0x00000002]    0xbfcbfbff

**Comment by Sibras** [comment]
I managed to track this one down as it appears it was a version inconsistency between the older asm gnutls code expects and the asm used in the project (gnutls pulls all its asm from openssl repo but is incompatible with any of the more recent asm versions). I pushed an update to the repo and updated the latest release on the release page with the appropriate fixes. Let me know if this corrects the problem for you.
